astronomer/airflow/version_check/update_checks.py

```diff
@@ -361,33 +387,105 @@ def available_update(self):
             rel_parsed_base_version = rel_parsed_version.major
 
             if rel_parsed_version > runtime_version and rel_parsed_base_version == base_version:
+                days_to_eol = None
+                if rel.end_of_support:
+                    now = utcnow()
+                    days_to_eol = (rel.end_of_support - now).days
+                dismissed_until = self.get_dismissed_until(rel.version)
                 return {
-                    "level": rel.level,
+                    "level": self.get_eol_level(days_to_eol),
                     "date_released": rel.date_released,
                     "description": rel.description,
                     "version": rel.version,
                     "url": rel.url,
                     "app_name": "Astronomer Runtime",
+                    "days_to_eol": days_to_eol,
+                    "dismissed_until": dismissed_until,
                 }
 
+            if not eol_warning_opt_out:
+                current_version = (
+                    session.query(AstronomerAvailableVersion)
+                    .filter(AstronomerAvailableVersion.version == str(runtime_version))
+                    .one_or_none()
+                )
+
+                if current_version and current_version.end_of_support:
+                    dismissed_eol_warning = (
+                        session.query(DismissedEOLWarning)
+                        .filter(
+                            DismissedEOLWarning.version == str(runtime_version),
+                            DismissedEOLWarning.dismissed_until > utcnow(),
+                        )
+                        .one_or_none()
+                    )
+
+                    if not dismissed_eol_warning:
+                        now = utcnow()
+                        days_to_eol = (current_version.end_of_support - now).days
+                        if days_to_eol <= 0:
+                            return {
+                                "level": self.get_eol_level(days_to_eol),
+                                "date_released": current_version.date_released,
+                                "description": "Current runtime version has reached its end of life!",
+                                "version": current_version.version,
+                                "url": current_version.url,
+                                "app_name": "Astronomer Runtime",
+                                "days_to_eol": days_to_eol,
+                                "dismissed_until": None,
+                            }
+                        elif days_to_eol <= self.eol_warning_threshold_days:
+                            return {
```

Comment:
Can we maybe refactor so there is less duplication between all these different scenarios? Coming in it's a bit hard to grok.

Reply:
Changed it completely and separated it for update notice and eol notice